### ISSUE TITLE
[BUG]:  运行GPT2-Gemini example，单卡和多卡并行显存占用没区别？

### ISSUE THREAD

**Issue by zhangyuanscall** [hpcaitech/ColossalAI#3281]
### 🐛 Describe the bug

# 以下为1卡与单机4卡的对比，显存占用并无差异，开启了tensor_parallelize，求问原因

### 单机4卡运行ColossalAI/examples/language/gpt/gemini 目录下的run_gemini.sh，bs为48，日志显示占用前向占用显存最大10G，但是Nvidai-smi显示占用20G，脚本设置如下

`
set -x
export DISTPLAN=${DISTPLAN:-"CAI_Gemini"}

export GPUNUM=${GPUNUM:-4}
export TPDEGREE=${TPDEGREE:-4}
export PLACEMENT=${PLACEMENT:-"cpu"}
export USE_SHARD_INIT=${USE_SHARD_INIT:-False}
export BATCH_SIZE=${BATCH_SIZE:-48}
export MODEL_TYPE=${MODEL_TYPE:-"gpt2_xl"}
export TRAIN_STEP=${TRAIN_STEP:-20}

echo -e "BATCH_SIZE: ${BATCH_SIZE}"
echo -e "TRAIN_STEP: ${TRAIN_STEP}"

if [ ${USE_SHARD_INIT} = "True" ]; then
  USE_SHARD_INIT="--shardinit"
else
  USE_SHARD_INIT=""
fi

mkdir -p gemini_logs

torchrun --standalone --nproc_per_node=${GPUNUM} ./train_gpt2.py \
--tp_degree=${TPDEGREE} \
--model_type=${MODEL_TYPE} \
--batch_size=${BATCH_SIZE} \
--placement=${PLACEMENT} \
${USE_SHARD_INIT} \
--distplan=${DISTPLAN} \
--train_step=${TRAIN_STEP} \
--shardinit \
2>&1 | tee ./gemini_logs/${MODEL_TYPE}_${DISTPLAN}_gpu_${GPUNUM}_bs_${BATCH_SIZE}_tp_${TPDEGREE}_${PLACEMENT}.log
`

#### 运行日志如下：

[03/28/23 18:39:16] INFO     colossalai - colossalai - INFO: ./train_gpt2.py:349 main
                    INFO     colossalai - colossalai - INFO: the size of testing model size is 469.9M.
Iter (loss=X.XXX):   0%|                                                                                                         | 0/53 [00:00<?, ?it/s][03/28/23 18:39:16] INFO     colossalai - colossalai - INFO: ./train_gpt2.py:334 main
                    INFO     colossalai - colossalai - INFO: After init optim, GPU memory usage: 12.03 MB, CPU memory usage: 4872.61 MB
[03/28/23 18:39:16] INFO     colossalai - colossalai - INFO: ./train_gpt2.py:349 main
                    INFO     colossalai - colossalai - INFO: the size of testing model size is 469.9M.
[03/28/23 18:39:16] INFO     colossalai - colossalai - INFO: ./train_gpt2.py:349 main
Iter (loss=X.XXX):   0%|                                                                                                         | 0/53 [00:00<?, ?it/s]                   INFO     colossalai - colossalai - INFO: the size of testing model size is 469.9M.
                    INFO     colossalai - colossalai - INFO: ./train_gpt2.py:349 main
                    INFO     colossalai - colossalai - INFO: the size of testing model size is 469.9M.
                    INFO     colossalai - colossalai - INFO: ./train_gpt2.py:350 main
                    INFO     colossalai - colossalai - INFO: After init model, GPU memory usage: 12.03 MB, CPU memory usage: 4872.61 MB
Iter (loss=X.XXX):   0%|                                                                                                         | 0/53 [00:00<?, ?it/s]`use_cache=True` is incompatible with gradient checkpointing. Setting `use_cache=False`...
`use_cache=True` is incompatible with gradient checkpointing. Setting `use_cache=False`...
`use_cache=True` is incompatible with gradient checkpointing. Setting `use_cache=False`...
`use_cache=True` is incompatible with gradient checkpointing. Setting `use_cache=False`...
Iter (loss=11.195):   0%|                                                                                                        | 0/53 [00:22<?, ?it/s][03/28/23 18:39:38] INFO     colossalai - colossalai - INFO: ./train_gpt2.py:380 train_step
                    INFO     colossalai - colossalai - INFO: [1/20] Forward GPU memory usage: 7880.17 MB, CPU memory usage: 5169.12 MB
Iter (loss=  nan):   0%|                                                                                                         | 0/53 [00:27<?, ?it/s]
[03/28/23 18:40:37] INFO     colossalai - colossalai - INFO: ./train_gpt2.py:391 train_step
                    INFO     colossalai - colossalai - INFO: [1/20] Backward GPU memory usage: 2129.11 MB, CPU memory usage: 5171.02 MB

#### Nvida-smi显示如下：
![image](https://user-images.githubusercontent.com/11641463/228211164-d9b2993e-a80e-44eb-9dcf-a669ddea41ef.png)

### 单机1卡运行ColossalAI/examples/language/gpt/gemini 目录下的run_gemini.sh，除GPUNUM=1，TPDEGREE=1，其他设置和单机4卡脚本相同，运行日志如下：
Iter (loss=11.188):   0%|▍                                                                                            | 1/209 [00:46<2:41:44, 46.65s/it]`use_cache=True` is incompatible with gradient checkpointing. Setting `use_cache=False`...
[03/28/23 18:46:11] INFO     colossalai - colossalai - INFO: ./train_gpt2.py:380 train_step
                    INFO     colossalai - colossalai - INFO: [1/20] Forward GPU memory usage: 7958.71 MB, CPU memory usage: 17371.55 MB
[03/28/23 18:46:55] INFO     colossalai - colossalai - INFO: ./train_gpt2.py:391 train_step
                    INFO     colossalai - colossalai - INFO: [1/20] Backward GPU memory usage: 2129.11 MB, CPU memory usage: 17371.75 MB
                    INFO     colossalai - colossalai - INFO: /opt/conda/lib/python3.7/site-packages/colossalai/nn/optimizer/zero_optimizer.py:218 step
                    INFO     colossalai - colossalai - INFO: Found overflow. Skip step
                    INFO     colossalai - colossalai - INFO: ./train_gpt2.py:397 train_step
                    INFO     colossalai - colossalai - INFO: [1/20] Optimizer step GPU memory usage: 2129.11 MB, CPU memory usage: 17371.85 MB
                    INFO     colossalai - colossalai - INFO: ./train_gpt2.py:402 train_step
                    INFO     colossalai - colossalai - INFO: [1/20] Loss:11.164, Step time: 57.732s, TFLOPS: 11.151, FWD time: 14.226s, BWD time:
                             43.491s, OPTIM time: 0.015s
Iter (loss=11.188):   1%|▉                                                                                            | 2/209 [01:44<3:03:27, 53.18s/it]

#### Nvidia-smi显示如下：

![image](https://user-images.githubusercontent.com/11641463/228212652-f28f518a-46b0-48d9-a208-55372f09bb4b.png)

### Environment

pip install  --no-cache-dir -i https://pypi.tuna.tsinghua.edu.cn/simple \  
    transformers==4.18.0 \
    torchmetrics==0.8.2 \
    torch==1.12.1+cu113 \
    torchvision==0.13.1+cu113 \
    torchaudio==0.12.1 --extra-index-url https://download.pytorch.org/whl/cu113

colossalai==0.2.7

**Comment by JThh** [comment]
Hi, since you are using `placement=cpu`, can you benchmark the CPU memory usage?
